src/devices/HCSR04/HCSR04.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.I2c.Drivers;
+using System.Device.Spi;
+using System.Device.Spi.Drivers;
+
+namespace Iot.Device.HCSR04
+{
+    public class Sonar : IDisposable
+    {
+        private readonly int _echo;
+        private readonly int _trigger;
+        private readonly GpioController _controller;
+
+        public Sonar(int triggerPin, int echoPin)
+        {
+            _echo = echoPin;
+            _trigger = triggerPin;
+            _controller = new GpioController();
+
+            _controller.OpenPin(_echo);
+            _controller.OpenPin(_trigger);
+            _controller.SetPinMode(_echo, PinMode.Input);
+            _controller.SetPinMode(_trigger, PinMode.Output);
+        }
+
+        /// <summary>
+        /// Gets the current distance in cm.
+        /// </summary>
+        public double GetDistance()
+        {
+            // Trigger input for 10uS to start ranging
+            _controller.Write(_trigger, PinValue.Low);
+            System.Threading.Thread.Sleep(TimeSpan.FromMilliseconds(1));
+            _controller.Write(_trigger, PinValue.High);
+            System.Threading.Thread.Sleep(TimeSpan.FromMilliseconds(0.01));
+            _controller.Write(_trigger, PinValue.Low);
+
+            // Start timers
+            Stopwatch stopTime = new Stopwatch(); stopTime.Start();
```

Comment:
Instead of starting the timers after initializing them, why not just use a do while loop instead, so that the body is ran at least once and the condition is checked at the end?